### PR TITLE
Mini Cart: Add an alternative icon setting 

### LINKED ISSUE #6868: Mini Cart > alternative icons
## Is your feature request related to a problem? Please describe.
follow up of https://github.com/woocommerce/woocommerce-blocks/issues/6617
image

Some improvements were recently made by @danielwrobert , to make the mini cart icon different from the one of the Cart Block.
Actually, only 1 icon is proposed in the mini cart block : it does not suite all contexts
examples : you make a jellery or fashion shop > you want a bag icon instead of a "supermarket like icon"

something like that
image

## Describe the solution you'd like
* add a bag icon (i don't think we need too much icons, but a more female icon, complementary from the male cart icon)
* permit to personalize the icon : link to replacement svg file, as proposed by @dashkevych ????

## Describe alternatives you've considered
N/A

## Additional context
N/A

### LINKED ISSUE #8556: Mini Cart block improvements
This is an epic to track all Mini Cart block related issues that we want to look into. At the beginning of the project, there were 25 open issues in WC Blocks + 3 issues in other repos:

These issues need engineering, product or design decisions, so it might be a good idea to surface the feedback request early in the cycle so we aren't blocked:

```[tasklist]
### Need design
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/8341
- [ ] https://github.com/woocommerce/woocommerce/issues/42435
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/7359
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/6868
```
```[tasklist]
### Ready to be worked on
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/8572
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/7176
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/8128
- [x] https://github.com/woocommerce/woocommerce-blocks/issues/7197
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/8603
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/6977
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/8148
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/8302
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/6257
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/6594
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/7016
- [ ] https://github.com/woocommerce/woocommerce/issues/42632
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/7750
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/8606
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/8787
- [ ] https://github.com/woocommerce/woocommerce/issues/42431
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/8814
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/8929
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/8946
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/8965
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/9044
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/9048
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/8716
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/8775
- [ ] https://github.com/woocommerce/woocommerce/issues/42645: ideally, I would prefer if Rubik worked on this issue given that it affects C&C extensibility points. We can keep it as a stretch goal in case we finish with everything else.
- [ ] https://github.com/woocommerce/woocommerce-blocks/pull/9051
```
```[tasklist]
### Require work on Gutenberg
- [ ] https://github.com/woocommerce/woocommerce/issues/42659
- [ ] https://github.com/woocommerce/woocommerce/issues/42482
```
```[tasklist]
### Require work on WooCommerce
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/6161
```
In theory, we shouldn't need to do anything related to these issues, @Aljullu will review them again before project end date:
```[tasklist]
### Repo cleanup (issues which might have been fixed before the project started)
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/7194
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/7522
- [ ] https://github.com/woocommerce/woocommerce-blocks/issues/6874
- [ ] https://github.com/Automattic/wp-calypso/issues/65662
- [ ] https://github.com/Automattic/wp-calypso/issues/66683
- [ ] https://github.com/Automattic/wp-calypso/issues/70204
```

### PR DESCRIPTION
**Pull request by danieldudzic**
This PR adds a Mini Cart block setting for alternative icons.

Fixes #6868

### Screenshots

| Before | After |
| ------ | ----- |
| Edit_Page_“Mini_cart”_‹_kirigami_—_WordPress-2|Edit_Page_“Mini_cart”_‹_kirigami_—_WordPress|

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

1. Add the Mini-Cart block to a page.
2. Select the Bag alternative icon and save.
3. Make sure it displays correctly in the editor and on the page.
4. Repeat the same for the third icon (Bag-Alt).

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Mini Cart: Add an alternative icon setting.

### CODE REVIEW

File: assets/js/blocks/mini-cart/block.tsx [hunk 1]
```diff
@@ -47,6 +47,7 @@ interface Props {
 	colorClassNames?: string;
 	style?: Record< string, Record< string, string > >;
 	contents: string;
+	miniCartIcon?: 'cart' | 'bag' | 'bag-alt';
```
Comment (imanish003) [inline assets/js/blocks/mini-cart/block.tsx]: We are using `'cart' | 'bag' | 'bag-alt'` in multiple places. Maybe we can consider creating a type for this?
```
type IconType = 'cart' | 'bag' | 'bag-alt';
```

What do you think?

File: assets/js/blocks/mini-cart/quantity-badge/index.tsx [hunk 2]
```diff
@@ -12,15 +12,23 @@ import './style.scss';
...
 }
 
-const QuantityBadge = ( { count }: Props ): JSX.Element => {
+const QuantityBadge = ( { count, icon }: Props ): JSX.Element => {
+	let miniCartIcon = cartOutline;
+	if ( icon === 'bag' ) {
+		miniCartIcon = bag;
+	} else if ( icon === 'bag-alt' ) {
+		miniCartIcon = bagAlt;
+	}
```
Comment (imanish003) [inline assets/js/blocks/mini-cart/quantity-badge/index.tsx]: Although it's not required but we may consider moving this logic to a function:

```

function getIcon(iconName: 'cart' | 'bag' | 'bag-alt') {
    switch (iconName) {
        case 'cart':
            return cartOutline;
        case 'bag':
            return bag;
        case 'bag-alt':
            return bagAlt;
        default:
            return cartOutline; // default case, could also throw an error
    }
}

```

**Review by imanish003 — APPROVED** [review]
Hi @danieldudzic, Great job! 🚀  Overall, PR looks good to me. I have left a few minor suggestions, but they are not critical, so I'm pre-approving the PR. Feel free to ignore those suggestions 🤝

**Review by Aljullu — COMMENTED** [review]
I didn't do a full review, but something I noticed is that the icons in this PR are  slightly smaller than what's in `trunk`. That's specially noticeable if you add the Mini-Cart block next to the Customer Account block:

Before | After
--- | ---
imatge | imatge

Can you reproduce as well? If so, I think this should be addressed before merging.

**Comment by danieldudzic** [comment]
> I didn't do a full review, but something I noticed is that the icons in this PR are slightly smaller than what's in `trunk`. That's specially noticeable if you add the Mini-Cart block next to the Customer Account block:
> Before 	After
> imatge 	imatge
> 
> Can you reproduce as well? If so, I think this should be addressed before merging.

Good catch! So this reduction in size should only apply to the icon in the block list?

cleanshot-2023-04-27-at-15 03 06402x

cc: @shaunandrews

**Comment by shaunandrews** [comment]
I gave this a quick spin; The selector for choosing between icons works as expect. Nice stuff!

--

As mentioned above, there are differing icons with differing sizes.

CleanShot 2023-06-01 at 12 08 34@2x

This is the block icon, which appears in the block inserter and the inspector sidebar, and is 24px in size.

--

CleanShot 2023-06-01 at 12 10 45@2x

These are the available icons, which appear on a users site alongside the Mini-Cart button, and are 32px in size.

**Comment by tarunvijwani** [comment]
Bumping it to the 10.5.0 release.

**Review by Aljullu — APPROVED** [review]
Code-wise it looks perfect, it's working as expected as well. My only last concern is that given that the icon now has some blank space around it, it causes the Mini-Cart button to be slightly bigger than it was. In some cases, pushing the entire header down a few pixels:

Before | After 
--- | ---
imatge | imatge

I think it could be solved by adding:

```CSS
.wc-block-mini-cart__icon {
  margin: -0.25em;
}
```

Before | After<br><small>(with CSS fix)</small>
--- | ---
imatge | imatge

Leaving the suggestion in case you think it's worth it. :slightly_smiling_face: Besides that, everything LGTM. Good job with this PR!

**Comment by danieldudzic** [comment]
> ```css
> .wc-block-mini-cart__icon {
>   margin: -0.25em;
> }
> ```

Thanks, @Aljullu - good catch again!

I'm usually quite uncomfortable using negative margins, but I have tested this, and it indeed works find, and I wasn't able to spot any issues.

**Pull request merged**